### ISSUE TITLE
ISSUE: cannot use a string pattern on a bytes-like object

### ISSUE THREAD

**Issue by eric1905** [moralmunky/Home-Assistant-Mail-And-Packages#853]
Currently I have the problem, that the integration is not starting.

**Environment (please complete the following information):**
 - OS: 10.5
 - Type of device: [e.g. RPi4
 - Home Assistant version:Home Assistant 2023.10.0
 - Hassio/Docker/Core? Supervisor 2023.10.0
 - Component version: 0.3.18

**Logs**

There is a lot of debug output before, but this seems to be ok so I am just pasting the last log output.
I am wondering about the outputs: 
Amazon Subject: Sxxxxxx Kxxxxxx wartet auf Deine Antwort. 
Amazon Subject: Andere IT-Entwickler bewerben sich für Circula: Customer Success Manager
Amazon Subject: b'Der monatliche Preis '
Because this is a mail from XING, LinkedIn and Spotify. Is this leading to the error? 

```
2023-10-06 16:45:02.020 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Attempting to find Amazon email with item list ...
2023-10-06 16:45:02.021 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Amazon email list: ['amazon.com', 'amazon.ca', 'amazon.co.uk', 'amazon.in', 'amazon.de', 'amazon.it', 'amazon.com.au', 'amazon.pl', 'amazon.es', 'amazon.fr', 'amazon.ae']
2023-10-06 16:45:02.021 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Amazon email search address: ['shipment-tracking@amazon.com', 'conferma-spedizione@amazon.com', 'confirmar-envio@amazon.com', 'versandbestaetigung@amazon.com', 'confirmation-commande@amazon.com']
2023-10-06 16:45:02.021 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] DEBUG subject: None
2023-10-06 16:45:02.021 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] DEBUG imap_search: (OR OR OR OR FROM "shipment-tracking@amazon.com" FROM "conferma-spedizione@amazon.com" FROM "confirmar-envio@amazon.com" FROM "versandbestaetigung@amazon.com" FROM "confirmation-commande@amazon.com" SINCE 03-Oct-2023)
2023-10-06 16:45:02.075 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] email_search value: ('OK', [b'759 762 763 764 777 779 780 787'])
2023-10-06 16:45:02.075 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Amazon emails found: 8
2023-10-06 16:45:02.123 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Email Multipart: True
2023-10-06 16:45:02.123 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Content Type: multipart/alternative
2023-10-06 16:45:02.123 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Amazon Subject: Sxxxxxx Kxxxxxx wartet auf Deine Antwort.
2023-10-06 16:45:02.124 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: will arrive:
2023-10-06 16:45:02.124 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: estimated delivery date is:
2023-10-06 16:45:02.125 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: guaranteed delivery date is:
2023-10-06 16:45:02.125 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: Arriving:
2023-10-06 16:45:02.125 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: Arriverà:
2023-10-06 16:45:02.125 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: arriving:
2023-10-06 16:45:02.125 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: Dostawa:
2023-10-06 16:45:02.125 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: Zustellung:
2023-10-06 16:45:02.125 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: Entrega:
2023-10-06 16:45:02.125 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: A chegar:
2023-10-06 16:45:02.125 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: Arrivée :
2023-10-06 16:45:02.182 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Email Multipart: True
2023-10-06 16:45:02.182 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Content Type: multipart/alternative
2023-10-06 16:45:02.183 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Amazon Subject: Andere IT-Entwickler bewerben sich für Circula: Customer Success Manager
2023-10-06 16:45:02.184 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: will arrive:
2023-10-06 16:45:02.185 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: estimated delivery date is:
2023-10-06 16:45:02.185 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: guaranteed delivery date is:
2023-10-06 16:45:02.185 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: Arriving:
2023-10-06 16:45:02.185 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: Arriverà:
2023-10-06 16:45:02.185 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: arriving:
2023-10-06 16:45:02.185 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: Dostawa:
2023-10-06 16:45:02.185 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: Zustellung:
2023-10-06 16:45:02.185 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: Entrega:
2023-10-06 16:45:02.185 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: A chegar:
2023-10-06 16:45:02.186 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Looking for: Arrivée :
2023-10-06 16:45:02.218 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Email Multipart: True
2023-10-06 16:45:02.219 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Content Type: multipart/alternative
2023-10-06 16:45:02.219 DEBUG (SyncWorker_2) [custom_components.mail_and_packages.helpers] Amazon Subject: b'Der monatliche Preis '
2023-10-06 16:45:02.224 ERROR (MainThread) [custom_components.mail_and_packages] Problem updating sensors: cannot use a string pattern on a bytes-like object
2023-10-06 16:45:02.224 ERROR (MainThread) [custom_components.mail_and_packages] Error fetching Mail and Packages (imap.gmx.net) data: cannot use a string pattern on a bytes-like object
2023-10-06 16:45:02.225 DEBUG (MainThread) [custom_components.mail_and_packages] Finished fetching Mail and Packages (imap.gmx.net) data in 3.571 seconds (success: False)
2023-10-06 16:45:02.225 ERROR (MainThread) [custom_components.mail_and_packages] Error updating sensor data: cannot use a string pattern on a bytes-like object
2023-10-06 16:45:02.225 WARNING (MainThread) [homeassistant.config_entries] Config entry 'imap.gmx.net' for mail_and_packages integration not ready yet: None; Retrying in background
```

**Comment by firstof9** [comment]
Looks like it's trying to process an attachment or something, I'll look into it.

**Comment by firstof9** [comment]
Should be fixed in `0.3.19`

**Comment by eric1905** [comment]
Looks good. Thank you.
